lib/Controller/BotController.php

```diff
@@ -149,6 +166,78 @@ public function sendMessage(string $token, string $message, string $referenceId
 		return new DataResponse([], Http::STATUS_CREATED);
 	}
 
+	#[BruteForceProtection(action: 'bot')]
+	#[PublicPage]
+	public function react(string $token, int $messageId, string $reaction): DataResponse {
+		try {
+			$bot = $this->getBotFromHeaders($token, $reaction);
+		} catch (\InvalidArgumentException $e) {
+			$response = new DataResponse([], $e->getCode());
+			if ($e->getCode() === Http::STATUS_UNAUTHORIZED) {
+				$response->throttle(['action' => 'bot']);
+			}
+			return $response;
+		}
+
+		$room = $this->manager->getRoomByToken($token);
+
+		$actorType = Attendee::ACTOR_BOTS;
+		$actorId = Attendee::ACTOR_BOT_PREFIX . $bot->getBotServer()->getUrlHash();
+
+		try {
+			$this->reactionManager->addReactionMessage(
+				$room,
+				$actorType,
+				$actorId,
+				$messageId,
+				$reaction
+			);
+		} catch (NotFoundException) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		} catch (ReactionAlreadyExistsException) {
+			return new DataResponse([], Http::STATUS_OK);
+		} catch (ReactionNotSupportedException | ReactionOutOfContextException | \Exception) {
+			return new DataResponse([], Http::STATUS_BAD_REQUEST);
+		}
+
+		return new DataResponse([], Http::STATUS_CREATED);
+	}
+
+	#[BruteForceProtection(action: 'bot')]
+	#[PublicPage]
+	public function deleteReaction(string $token, int $messageId, string $reaction): DataResponse {
+		try {
+			$bot = $this->getBotFromHeaders($token, $reaction);
+		} catch (\InvalidArgumentException $e) {
+			$response = new DataResponse([], $e->getCode());
+			if ($e->getCode() === Http::STATUS_UNAUTHORIZED) {
+				$response->throttle(['action' => 'bot']);
+			}
+			return $response;
+		}
+
+		$room = $this->manager->getRoomByToken($token);
+
+		$actorType = Attendee::ACTOR_BOTS;
+		$actorId = Attendee::ACTOR_BOT_PREFIX . $bot->getBotServer()->getUrlHash();
+
+		try {
+			$this->reactionManager->deleteReactionMessage(
+				$room,
+				$actorType,
+				$actorId,
+				$messageId,
+				$reaction
+			);
+		} catch (ReactionNotSupportedException | ReactionOutOfContextException | NotFoundException) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		} catch (\Exception) {
```

Comment:
Maybe we should catch `\Throwable` here?

Reply:
If it's not expected, don't catch it? :P

Reply:
Agree, but…
Already a lot of things can be caught with `Exception`. Are all of them expected?